tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordEntry.cs

```diff
@@ -26,16 +26,54 @@ public class RecordEntry
 
         public RequestOrResponse Response { get; set; } = new RequestOrResponse();
 
-        public string RequestUri { get; set; }
+        private string _requestUri;
+        public string RequestUri
+        {
+            get { return this._requestUri; }
+            set {
+                // If the requestUri is being modified, set the flag to true
+                if (DebugLogger.CheckLogLevel(LogLevel.Debug) && this._requestUri != value) 
+                {
+                    requestUriIsModified = true;
+                } 
+                this._requestUri = value; 
+            }
+        }
 
         public bool IsTrack1Recording { get; set; }
 
         public RequestMethod RequestMethod { get; set; }
 
         public int StatusCode { get; set; }
+        // Flag to indicate if the requestUri has been modified
+        public bool requestUriIsModified { get; set; } = false;
+
+        /// <summary>
+        /// Checks if the RecordEntry instance or any of its properties have been modified.
+        /// Primarily used to determine if the RecordEntry has been sanitized.
+        /// </summary>
+        /// <returns>True if any modification has been made, otherwise false.</returns>
+        public bool isModified()
+        {
+            return this.requestUriIsModified || this.Request.IsModified.Headers || this.Request.IsModified.Body || this.Response.IsModified.Headers || this.Response.IsModified.Body;
+        }
 
-        public static RecordEntry Deserialize(JsonElement element)
+        /// <summary>
+        /// Initializes the IsModified flags for the RecordEntry instance.
+        /// Primarily used to determine if the RecordEntry has been sanitized.
+        /// </summary>
+        public void initializeIsModifiedFlag()
         {
+            if (DebugLogger.CheckLogLevel(LogLevel.Debug))
+            {
+                this.requestUriIsModified = false;
+                this.Request.IsModified = new RequestOrResponse.RequestOrResponseIsModified();
+                this.Response.IsModified = new RequestOrResponse.RequestOrResponseIsModified();
+            }
+        }
+
+        public static RecordEntry Deserialize(JsonElement element)
+            {
             var record = new RecordEntry();
```

Comment:
We should update `Deserialize` to set `Request.IsModified` and `requestUriIsModified` if putting the defaults on the property directly won't work for you.

Reply:
As discussed offline, we want this resetting flags to be as close as possible to Sanitization.
Before sanitization - we take it to be a clean slate
After sanitization - we get the changes from what the flags say